src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1

```diff
@@ -1763,106 +1763,101 @@ class Organization: ADOSVTBase
-            if($TotalPCAMembers -gt 0){
-                $controlResult.AddMessage("Verify the following Project Collection Administrators: ")
-                $controlResult.AdditionalInfo += "Total number of Project Collection Administrators: " + $TotalPCAMembers;
-            }
-
-            if (($SvcAndHumanAccounts.humanAccount | Measure-Object).Count -gt 0) {
-                $humanAccounts = $SvcAndHumanAccounts.humanAccount | Select-Object displayName, mailAddress
-                $controlResult.AddMessage("`nHuman Administrators: $(($humanAccounts| Measure-Object).Count)", $humanAccounts)
-                $controlResult.SetStateData("List of human Project Collection Administrators: ",$humanAccounts)
+        $controlResult.VerificationResult = [VerificationResult]::Failed
+        try{
+            $TotalPCAMembers=0
+            
+            if ($this.PCAMembersList.Count -eq 0) {
+                $this.PCAMembersList =@([AdministratorHelper]::GetTotalPCAMembers($this.OrganizationContext.OrganizationName))
+            }
+            $PCAMembers = $this.PCAMembersList
+            $TotalPCAMembers = $PCAMembers.Count
+            $controlResult.AddMessage("There are a total of $TotalPCAMembers Project Collection Administrators in your organization.")
+            if ($this.graphPermissions.hasGraphAccess)
+            {  
+                $SvcAndHumanAccounts = @([IdentityHelpers]::DistinguishHumanAndServiceAccount($PCAMembers, $this.OrganizationContext.OrganizationName))
+                $humanAccounts= @($SvcAndHumanAccounts.humanAccount | Select-Object displayName, mailAddress)
+                $svcAccounts= @($SvcAndHumanAccounts.serviceAccount | Select-Object displayName, mailAddress)
+                if($humanAccounts.Count -gt $this.ControlSettings.Organization.MaxPCAMembersPermissible){
+                    $controlResult.AddMessage([VerificationResult]::Failed,"Number of human administrators configured are more than the approved limit: $($this.ControlSettings.Organization.MaxPCAMembersPermissible)");
+                }
+                else{
+                    $controlResult.AddMessage([VerificationResult]::Passed,"Number of human administrators configured are within the approved limit: $($this.ControlSettings.Organization.MaxPCAMembersPermissible)");
+                }
+                [AdministratorHelper]::PopulatePCAResultsToControl($humanAccounts, $svcAccounts, $controlResult)
             }
-
-            if (($SvcAndHumanAccounts.serviceAccount | Measure-Object).Count -gt 0) {
-                $svcAccounts = $SvcAndHumanAccounts.serviceAccount | Select-Object displayName, mailAddress
-                $controlResult.AddMessage("`nService Account Administrators: $(($svcAccounts| Measure-Object).Count)", $svcAccounts)
-                $controlResult.SetStateData("List of service account Project Collection Administrators: ",$svcAccounts)
+            #ToDo Add warning that control was evaluated without graph access
```

Comment:
todo?

Reply:
Manish asked to add for Sourabh's reference. Changed it to add more details just like Himanshu did.